ambry-store/src/main/java/com.github.ambry.store/CompactionLog.java

```diff
@@ -313,13 +327,13 @@ static CycleLog fromBytes(DataInputStream stream, StoreKeyFactory storeKeyFactor
      */
     Phase getPhase() {
       Phase phase;
-      if (copyStartTime == -1) {
+      if (copyStartTime == UNINITIALIZED_TIMESTAMP) {
         phase = Phase.PREPARE;
-      } else if (switchStartTime == -1) {
+      } else if (commitStartTime == UNINITIALIZED_TIMESTAMP) {
         phase = Phase.COPY;
-      } else if (cleanupStartTime == -1) {
-        phase = Phase.SWITCH;
-      } else if (cycleEndTime == -1) {
+      } else if (cleanupStartTime == UNINITIALIZED_TIMESTAMP) {
+        phase = Phase.COMMIT;
+      } else if (cycleEndTime == UNINITIALIZED_TIMESTAMP) {
         phase = Phase.CLEANUP;
       } else {
         phase = Phase.DONE;
```

Comment:
Yes - was talking about going from DONE of one cycle to PREPARE of next cycle. You are saying that for every new cycle, a new object of `CycleLog` will be instantiated?